docs/getting-started/basics.md

```diff
@@ -0,0 +1,210 @@
+    )
+```
+
+If you need a component not provided by Reflex, you can check the [3rd party ecosystem]({custom_components.path}) or [wrap your own React component]({docs.wrapping_react.guide.path}).
+
+
+## Customizing and styling components
+
+Components can be customized using [props]({docs.components.props.path}), which are passed in as keyword arguments to the component function.
+
+Each component has props that are specific to that component. Check the docs for the component you are using to see what props are available.
+
+```python demo exec
+def half_filled_progress():
+    return rx.progress(value=50)
+```
+
+In addition to component-specific props, components can also be styled using CSS properties passed as props.
+
+```python demo exec
+def round_button():
+    return rx.button("Click Me", border_radius="15px", font_size="18px")
+```
+
+See the [styling guide]({docs.styling.overview.path}) for more information on how to style components
+
+## Making components interactive
+
+Apps often need to respond to user input or update the screen based on some event. Reflex handles this through [state]({docs.state.overview.path}), which is a Python class that stores variables that can change when the app is running, as well as the functions that can change those variables.
+
+To define a state class, subclass `rx.State` and define fields that store the state of your app. The state vars should have a type annotation, and can be initialized with a default value.
+
+```python
+class MyState(rx.State):
+    count: int = 0
+```
+
+## Referencing state in components
+
+To reference a state var in a component, you can pass it as either a prop or a child of the component.
+
+```python demo exec
+class MyState(rx.State):
+    count: int = 0
+
+def counter():
+    return rx.hstack(
+        rx.heading("Count: "),
+        # Reference the state var in the component
+        rx.heading(MyState.count),
+    )
+```
+
+Vars can be referenced in multiple components, and will automatically update when the state changes.
+
+## Triggering events
+
+Components have special props, called event triggers, that can be used to make components interactive. Event triggers are defined as special component props, such as `on_click`, and can be used to trigger event handlers.
+
+
+```python demo exec
+def toast_button():
+    return rx.button("Show toast", on_click=rx.toast("Hello World"))
+```
+
+Event handlers are functions that update the state in response to events. They are defined as methods in the `State` class, and can be called by event triggers, or by other event handlers.
+
+```python demo exec
+class EventState(rx.State):
+    message: str = "Hello World"
+
+    def update_message(self):
+        self.message = "Goodbye World"
+
+def event_button():
+    return rx.button("Update Message", on_click=EventState.update_message)
+
+def index():
+    return rx.box(
+        rx.text(EventState.message),
+        event_button()
+    )
+```
+
+
+## Updating the screen
```

Comment:
this should go above the last example where we actually use the event handler to update a state var that will update the screen